components/omega/doc/design/Metadata.md

```diff
@@ -0,0 +1,408 @@
+itself to enable appropriate metadata in all model output. The capabilities
+described here must interact with other parts of the I/O within OMEGA,
+particularly the lower-level I/O interfaces that
+perform the actual reading/writing to disk and the management of
+files or streams and their contents.
+
+## 2 Requirements
+
+### 2.1 Requirement: Data types
+
+Metadata can be in all supported OMEGA data types (I4, I8, R4, R8,
+Boolean/logical, strings). Initially only scalar metadata
+is required, but extending to vectors/arrays may be desireable.
+
+### 2.2 Requirement: Global and variable metadata
+
+Metadata refers not only to field/variable metadata but
+also metadata associated with the model, code base, simulation
+or data file. This capability must be able to track metadata
+for these global/general metadata in addition to metadata
+attached to particular fields or variables.
+
+### 2.3 Requirement: Metadata conventions
+
+Where possible, all metadata must conform to the climate/forecast (CF)
+metadata conventions at https://cfconventions.org. 
+
+### 2.4 Requirement: Required metadata
+
+To comply with conventions and E3SM practices, there will be a minimum
+set of required metadata and interfaces must enforce this minimum set.
+This minimum will be defined later, but for variables, this would
+typically include a name (short), units, long name/description,
+standard CF name (if exists), _FillValue, valid min/max and dimensions.
+
+### 2.5 Requirement: Dimensions
+
+For array variables, a definition of each dimension used is required.
+In an unstructured model like OMEGA, the dimension describes the
+global extent of the index space for each dimension that is then paired
+with mesh fields for the full description of OMEGA coordinate locations.
+For time dimensions, we require support for both a fixed length time
+dimension (eg to support multiple time levels in a restart) as well
+as an unlimited dimension to support time series.
+
+### 2.6 Requirement: Available fields
+
+To support other I/O components, we require a means for maintaining
+a list of fields available for I/O based on the requested model
+configuration. This enables the I/O streams to define contents of
+a given file as part of model configuration input by supplying a
+list of variable names to be included.
+
+### 2.7 Desired: Metadata groups
+
+Because lists of contents in the model config file can get long,
+it would be useful to define and maintain a metadata group that
+is simply a shorthand to a set of common fields (eg meshFields or
+prognosticVars). Note that these would only be for the purpose
+of shortening lists in the config file or managing lists internally.
+The full list of fields would still appear in output file metadata.
+
+## 3 Algorithmic Formulation
+
+No algorithms are used.
+
+## 4 Design
+
+The general philosophy in this design is to have modules that
+"own" the variable define the field metadata and add it to a
```

Comment:
```suggestion
"own" the variable, define the field metadata and add it to a
```
Is a comma missing here?

Reply:
@cbegeman Actually, no.  In this case "that own the variable" was meant to be a restrictive clause modifying "modules", but I can see the point of confusion.  Maybe clearer if I write "The general philosophy is that modules will define the metadata for each field they own and add it..."  Thanks for re-reviewing and I'll make that change after seeing what other changes are suggested.